api/README.md

```diff
@@ -100,3 +100,55 @@ The high-level structure of the `v4` manifest is as follows:
 - For `reftest` entries: `[url, references, extras]`
   - `references` example: `[[reference_url1, "=="], [reference_url2, "!="], ...]`
   - `extras` example: `{"timeout": "long", "viewport_size": ..., "dpi": ...}`
+
+## /api/results/upload
+
+Uploads a wptreport to the dashboard to create the test run.
+
+This endpoint only accepts POST requests, and need to be authenticated via HTTP basic auth. Please
+contact [Ecosystem Infra](mailto:ecosystem-infra@chromium.org) if you want to register as a "test
+runner" to upload results.
+
+### File payload
+
+__Content type__: `multipart/form-data`
+
+__Parameters__
+
+__`result_file`__: A gzipped JSON file produced by `wpt run --log-wptreport`.
+
+The JSON file roughly looks like this:
+
+```json
+{
+  "results": [...],
+  "run_info": {
+    "revision": "WPT revision of the test run",
+    "product": "your browser",
+    "browser_version": "version of the browser",
+    "os": "your os",
+    "os_version": "OPTIONAL OS version",
+    ...
+  }
+}
+```
+
+Note that `run_info.{revision,product,browser_version,os}` are required, and should be automatically
+generated by `wpt run`. If for some reason the report does not contain these fields (e.g. old WPT
+version, Sauce Labs, or custom runners), they can be overridden with the following *optional*
+parameters in the POST payload (this is NOT recommended; please include metadata in the reports
+whenever possible):
```

Comment:
I can feel your distaste in these words :) If this is only for the sake of the current results reporter, remove it! It's looking like it's easier for me to insert this information into the report itself, anyway

Reply:
I provided this override mostly for debugging (before I added the metadata to wptreport). And as you can see, the inconsistency of names is not great...

Yeah it should be easy for a custom wptrunner to add these fields. The only concern for now is the Sauce Labs. Ideally, reports generated by the official `wpt run` should be compatible with the receiver. Let me see if I can fix it from wptrunner.

Reply:
> Let me see if I can fix it from wptrunner.

Only if you're feeling bored. Remember that consolidating multiple reports, so I'm already responsible for manually building this data structure. It's really simple for me to insert the fields that are missing--honest!